web/src/engine/websites/ManhwaWeb.ts

```diff
@@ -0,0 +1,80 @@
+import { Tags } from '../Tags';
+import icon from './ManhwaWeb.webp';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON } from '../platform/FetchProvider';
+
+type APIMangas = {
+    data: APIManga[],
+    next: boolean
+}
+
+type MangaResult = {
+    mangas: Manga[],
+    next: boolean
+}
+type APIManga = {
+    _id: string,
+    the_real_name: string,
+    chapters: APIChapter[]
+}
+
+type APIChapter = {
+    link: string,
+    chapter: number,
+    img: string[]
+}
+
+type APIPages = {
+    chapter: APIChapter
+}
+
+@Common.ImageAjax(undefined, true)
+export default class extends DecoratableMangaScraper {
+    public readonly apiUrl = 'https://manhwawebbackend-production.up.railway.app';
+
+    public constructor() {
+        super('manhwaweb', 'MahwaWeb', 'https://manhwaweb.com', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Media.Manga, Tags.Media.Novel, Tags.Language.Spanish, Tags.Source.Aggregator, Tags.Rating.Pornographic);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExpSafe(`^${this.URI.origin}/(manga|manhwa)/[^/]+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const slug = url.split('/').pop();
+        const { the_real_name } = await FetchJSON<APIManga>(new Request(new URL(`/manhwa/see/${slug}`, this.apiUrl)));
+        return new Manga(this, provider, slug, the_real_name);
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList: Manga[] = [];
+        for (let page = 1, run = true; run; page++) {
+            const { mangas, next } = await this.GetMangasFromPage(page, provider);
+            mangaList.push(...mangas);
+            run = next;
+        }
+        return mangaList;
+    }
+
+    private async GetMangasFromPage(page: number, provider: MangaPlugin): Promise<MangaResult> {
+        const { data, next } = await FetchJSON<APIMangas>(new Request(new URL(`/manhwa/library?estado=&tipo=&erotico=&demografia=&page=${page}`, this.apiUrl)));
```

Comment:
Can't the query parameters `estado=&tipo=&erotico=&demografia=` without an explicit value be removed from the URL?

Reply:
Nope. There was way more. Removing one of those i kept break the request.